### PR TITLE
Improve CB accuracy for Avc2

### PR DESCRIPTION
**Pull request by RipleyTom**
Some inaccuracies in the way we handle CBs led to some issues for DBZ: RB2.
This emulates the behaviour of the internal Avc2 dispatcher.

### CODE REVIEW

File: rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp [hunk 1]
```diff
@@ -434,6 +434,13 @@ error_code cellSysutilAvc2Unload()
 		return CELL_AVC2_ERROR_NOT_INITIALIZED;
 	}
 
+	// Dragon Ball: Raging Blast 2 expects this despite despite cellSysutilAvc2Unload supposedly not passing that event
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp:437]: Despite twice

File: rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp [hunk 2]
```diff
@@ -110,10 +115,42 @@ struct avc2_settings
...
+					if ((event == CELL_AVC2_EVENT_LOAD_FAILED ||
+							event == CELL_AVC2_EVENT_UNLOAD_SUCCEEDED ||
+							event == CELL_AVC2_EVENT_UNLOAD_FAILED) &&
+						error_code < 2)
+					{
+						sysutil_unregister_cb_with_id<avc2_cb_handle_t>();
+
+						std::lock_guard lock(this->mutex_cb);
+						this->avc2_cb = vm::null;
+						this->avc2_cb_arg = vm::null;
```
Comment (elad335) [inline rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp:146]: Shouldn't it empty the args before calling the callback? (in the previous lock scope)

Comment (RipleyTom) [reply rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp:146]: It's accurate to the dispatcher of avc2(except for the addition of the mutex which I added for sanity sake)

File: rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp [hunk 3]
```diff
@@ -1062,31 +1069,7 @@ error_code cellSysutilAvc2UnloadAsync2(u32 mediaType)
 		return CELL_AVC2_ERROR_INVALID_ARGUMENT;
 
 	auto& settings = g_fxo->get<avc2_settings>();
-
-	if (settings.avc2_cb)
-	{
-		if (mediaType == CELL_SYSUTIL_AVC2_VOICE_CHAT)
-		{
-			// TODO: signal CELL_AVC2_EVENT_UNLOAD_FAILED instead if the video chat is still loaded
-		}
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp:1071]: why did you remove this?

Comment (RipleyTom) [reply rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp:1071]: Because it's not in the function...

File: rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp (hunk 2 above)
Comment (Megamouse) [reply rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp:146]: I dont think that it resets these if you only unload the video chat.

Comment (Megamouse) [reply rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp:146]: I mean the entire unregister and reset of the callbacks

Comment (Megamouse) [reply rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp:146]: There is no video callback and you can keep using the voice chat after unloading the video chat

Comment (RipleyTom) [reply rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp:146]: I changed it to return 2 for the error when unloading video which is probably why that check is there in the dispatcher.

Comment (Megamouse) [reply rpcs3/Emu/Cell/Modules/cellSysutilAvc2.cpp:146]: I kinda still doubt that it's the correct solution, but I guess it's good enough for now
